lms/views/dashboard.py

```diff
@@ -109,8 +114,12 @@ def get_request_assignment(self):
         return assignment
 
     def _set_lti_user_cookie(self, response):
+        lti_user = self.request.lti_user
+        if not lti_user:
```

Comment:
This conditional should fix the second scenario in https://github.com/hypothesis/lms/pull/6209#discussion_r1576145538